### ISSUE TITLE
Telegraf image does not include config directory

### ISSUE THREAD

**Issue by Hipska** [influxdata/influxdata-docker#568]
Although the directory `/etc/telegraf/telegraf.d` directory is created in the image, default usage of the image does not load config files in this directory.

**Comment by Hipska** [comment]
@powersj something that could be added when releasing new version?

**Comment by powersj** [comment]
The docker images run `telegraf` when they are launched. In this case, Telegraf looks in the following places for a configuration file default:

1. `$TELEGRAF_CONFIG_PATH`
2. `/home/telegraf/.telegraf/telegraf.conf`
3. `/etc/telegraf/telegraf.conf`

The image instructions are primarily centered around updating the one configuration file or uploading one configuration file.

Users can pass additional arguments to telegraf as well, like `--config-directory`. I would expect if you wanted this behavior you would load all your config files in the common directory and then add the directory option:

```s
docker run \
    -v $PWD/empty.toml:/etc/telegraf/telegraf.conf  \
    -v $PWD/config01.toml:/etc/telegraf/telegraf.d/telegraf01.conf:ro \
    -v $PWD/config02.toml:/etc/telegraf/telegraf.d/telegraf02.conf:ro \
    telegraf --config-directory /etc/telegraf/telegraf.d
```

As far as adding `--config-directory /etc/telegraf/telegraf.d` as a default configuration setting, my concern is changing the behavior of these images. We learned with the `root` to `telegraf` user that images are heavily used and users already take advantage of many of the settings and options. For a basic config, it would not change anything, but users with more advanced configs could result in errors to their configs.

As such I am inclined to say this use case should use the available options that I outlined above.

**Comment by Hipska** [comment]
Hmm, 3 remarks to that:

1. Users that are following those image instructions won’t have any files in that directory.
2. Users that do have files in there would already have a custom command that specifies to use that directory.
3. Why is it added to the image if it isn’t meant to be used. Better to not add unused stuff for keeping image clean and small. 

For both scenarios 1 and 2 I see no impact for existing users. New users will have benefits from it in the fact they don’t need to override the command and the image behaves the same as you would install telegraf as a service via the distro repositories. (Rpm, deb, …)

**Comment by powersj** [comment]
> Why is it added to the image if it isn’t meant to be used. Better to not add unused stuff for keeping image clean and small.

What is "it"? The configuration directory? If so, the directory is added because we use the DEB to install Telegraf, which creates it. I also reject the assertion that it is not meant to be used. Nowhere do we say that and clearly the above shows how it can be used.

I specifically called out that the docker image runs `telegraf` when it runs. This is opposed to the DEBs and RPMs which use the following command:

```s
/usr/bin/telegraf -config /etc/telegraf/telegraf.conf -config-directory /etc/telegraf/telegraf.d $TELEGRAF_OPTS
```

You have identified that the behavior is different, but that alone is not a reason to change this. Sure, this makes it easier for new users, but we have hundreds/thousands/millions of existing users and deployments. I would want further testing of various scenarios before we change that long-standing behavior to ensure there are no regressions for our users.

If this change is really interesting then I think at the very least these types of tests should occur:

* Update the entry point to add the config directory option and build the container to ensure it works with `setpriv` and `su-exec`
* Test launching a container with and without files in that directory
* Test launching the container with other config options to ensure they work (e.g. --config, --help)
* Test running the container as a different user
* Test running with another config directory added as a docker run option
* Delete the directory and ensure Telegraf starts

**Comment by Hipska** [comment]
Testing is good and should indeed be done, but most of the ones you mention don’t relate to the fact it is run inside a container and should be tested with telegraf itself. 

Also, why is the .deb used? Wouldn’t it make more sense to use the tar? With that you can select which files you want to extract. The deb also installs systemd and logrotate services, which also don’t make much sense in this context.

**Comment by Hipska** [comment]
See also https://github.com/influxdata/telegraf/issues/5656

**Comment by Hipska** [comment]
@powersj any feedback?

**Comment by networkandcode** [comment]
Same issue for me too with telegram:1.23.2-alpine

**Comment by Hipska** [comment]
@powersj any feedback for this almost 1 year old issue/comments?

**Comment by Hipska** [comment]
Seems to be finally solved with 1.26! https://github.com/influxdata/telegraf/pull/12608
